### PR TITLE
TINY-6870: Switch Nonbreaking plugin to use BDD style tests

### PR DESCRIPTION
**Pull request by JamesToohey**
Related Ticket: TINY-6870

Description of Changes:
Switch `nonbreaking` plugin to use BDD style tests

Pre-checks:
* [x] ~Changelog entry added~
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
-        Step.sync(() => {
-          const args = editor.fire('keydown', { keyCode: VK.TAB });
-          Assert.eq('Default should be prevented', true, args.isDefaultPrevented());
-          Assert.eq('Should not propagate', true, args.isImmediatePropagationStopped());
-        })
-      )
-    ], onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.nonbreaking.NonbreakingForceTabTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
```
Comment (lnewson) [inline modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts:12]:
```suggestion
  const hook = TinyHooks.bddSetupLight<Editor>({
```
This was using a light setup previously.

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts [hunk 2]
```diff
@@ -1,42 +1,35 @@
...
     theme: 'silver',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
-}
-);
+  }, [ Theme, Plugin ], true);
+
+  it('TBA: Undo level on insert tab', () => {
+    const editor = hook.editor();
+    editor.focus();
```
Comment (lnewson) [inline modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts:22]: Is this needed given you've put this in the bdd setup now?

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts [hunk 3]
```diff
@@ -1,42 +1,35 @@
...
+    Keyboard.activeKeystroke(TinyDom.document(editor), Keys.tab());
```
Comment (lnewson) [inline modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts:23]: Thanks for swapping this from using the older `editor.fire` version.

**Review by lnewson — APPROVED** [review]

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts (hunk 3 above)
Comment (JamesToohey) [reply modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingForceTabTest.ts:22]: I still have a test using the `editor.fire` - should I remove it?

File: modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingSettingsTest.ts [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
 import Editor from 'tinymce/core/api/Editor';
 import EditorManager from 'tinymce/core/api/EditorManager';
 import * as Settings from 'tinymce/plugins/nonbreaking/api/Settings';
 
-UnitTest.test('browser.tinymce.plugins.nonbreaking.SettingsTest', () => {
-  Assertions.assertEq('TestCase-TBA: NonBreaking: Should be 0 as default', 0, Settings.getKeyboardSpaces(new Editor('x', {}, EditorManager)));
-  Assertions.assertEq('TestCase-TBA: NonBreaking: Should return 3 if set to true', 3, Settings.getKeyboardSpaces(new Editor('x', { nonbreaking_force_tab: true }, EditorManager)));
-  Assertions.assertEq('TestCase-TBA: NonBreaking: Should return 0 if set to false', 0, Settings.getKeyboardSpaces(new Editor('x', { nonbreaking_force_tab: false }, EditorManager)));
-  Assertions.assertEq('TestCase-TBA: NonBreaking: Should return number if set', 4, Settings.getKeyboardSpaces(new Editor('x', { nonbreaking_force_tab: 4 }, EditorManager)));
+describe('browser.tinymce.plugins.nonbreaking.SettingsTest', () => {
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/nonbreaking/test/ts/browser/NonbreakingSettingsTest.ts:8]:
```suggestion
describe('browser.tinymce.plugins.nonbreaking.NonbreakingSettingsTest', () => {
```

**Review by ltrouton — APPROVED** [review]

**Review by HAFRMO — APPROVED** [review]

**Pull request merged**